src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx

```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
     const [username, setUsername] = useState(getUsername)
     //#endregion
 
+    //#region paste status
+    const [pastedStatus, setPastedStatus] = useState<boolean | 'undetermined'>('undetermined')
+    //#ednregion
+
+    //#region create post status
+    const [createStatus, setCreateStatus] = useState<boolean | 'undetermined'>('undetermined')
+    //#endregion
+
     const onNext = async () => {
-        currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
-            status: 'during',
-            username: username,
-            persona: persona.toText(),
-        } as ImmersiveSetupCrossContextStatus)
-        const connecting = new ProfileIdentifier(ui.networkIdentifier, username)
-        await Services.Identity.attachProfile(connecting, persona, { connectionConfirmState: 'confirmed' })
-        ui.taskGotoProfilePage(connecting)
-        await sleep(400)
-        setStep(SetupGuideStep.PasteIntoBio)
+        switch (step) {
+            case SetupGuideStep.FindUsername:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
+                    status: SetupGuideStep.PasteIntoBio,
+                    username,
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                const connecting = new ProfileIdentifier(ui.networkIdentifier, username)
+                await Services.Identity.attachProfile(connecting, persona, { connectionConfirmState: 'confirmed' })
+                ui.taskGotoProfilePage(connecting)
+                setStep(SetupGuideStep.PasteIntoBio)
+                break
+            case SetupGuideStep.PasteIntoBio:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
```

Comment:
Use `json-stable-stringify` in this case or a unintented infinity loop might comes in.